Tests/Marketplace/marketplace_services.py

```diff
@@ -1475,49 +1476,56 @@ def prepare_release_notes(self, index_folder_path, build_number, pack_was_modifi
                             changelog[latest_release_notes] = version_changelog
 
                         if modified_release_notes_lines_dict:
-                            logging.info("updating changelog entries for modified rn")
+                            logging.info("Updating changelog entries for modified release notes")
                             for version, modified_release_notes_lines in modified_release_notes_lines_dict.items():
                                 updated_entry = self._get_updated_changelog_entry(
                                     changelog, version, release_notes=modified_release_notes_lines)
                                 changelog[version] = updated_entry
 
-                else:  # will enter only on initial version and release notes folder still was not created
-                    if len(changelog.keys()) > 1 or Pack.PACK_INITIAL_VERSION not in changelog:
+                else:
+                    if len(changelog.keys()) > 1:
+                        # If there is no release notes dir but the changelog has a few entries in it,
+                        # there is a mismatch
                         logging.warning(
                             f"{self._pack_name} pack mismatch between {Pack.CHANGELOG_JSON} and {Pack.RELEASE_NOTES}")
                         task_status, not_updated_build = True, True
                         return task_status, not_updated_build
 
-                    changelog[Pack.PACK_INITIAL_VERSION] = self._create_changelog_entry(
-                        release_notes=self.description,
-                        version_display_name=Pack.PACK_INITIAL_VERSION,
-                        build_number=build_number,
-                        initial_release=True,
-                        new_version=False)
+                    else:
```

Comment:
Redundant `else` after a `return`.

Reply:
I know, but I find it to help the readability of the code, as this function has many `if` statements. Adding this helped structurize the conditions in to logical cases. But your call